tests/test_reference_langevin_integrator.py

```diff
@@ -52,6 +53,32 @@ def test_reference_langevin_integrator(threshold=1e-4):
         assert histogram_mse < threshold
 
 
+def test_reference_langevin_integrator_deterministic():
+    """
+    Asserts that trajectories produced by `_deterministic` methods
+    are reproducible, and that the implementation in terms of the XLA
+    scan primitive gives a consistent result to that of the
+    implementation as a Python for-loop
+    """
+    force_fxn = lambda x: -4 * x ** 3
+    langevin = LangevinIntegrator(force_fxn, masses=1.0, temperature=300.0, dt=0.1, friction=1.0)
+    x0, v0 = 0.1 * np.ones((2, 5))
+
+    # private "reference" implementation (not using XLA primitives)
+    xs1, vs1 = langevin._multiple_steps_deterministic(jax.random.PRNGKey(1), x0, v0)
```

Comment:
Is there value having the non-deterministic implementation? Seems like it would be better to have determinism all the time than to have two implementations, especially if the deterministic one can be made faster than the non-deterministic one.

Reply:
I like the idea of making everything deterministic, but we haven't fully bought in to the `jax.random` API, and as it is removing the non-deterministic version would break a fair amount of existing code.

Reply:
Perhaps modify the constructor to take a seed and either use the jax random API or the numpy random depending on the implementation. That will make it inline with our C++ Langevin integrator and give us determinism.

Reply:
> Perhaps modify the constructor to take a seed and either use the jax random API or the numpy random depending on the implementation

I like the idea of making the jax reference `LangevinIntegrator` more consistent with the C++ implementation, but I think I'd actually prefer modifying the signature of the C++ version to accept `seed` as an argument to `step_fwd` (and `Context::multiple_steps`), thus moving away from keeping RNGs as hidden internal state of objects. One case where this could be useful is sampling independent chains, where we could avoid initializing a new integrator object for each chain.

To make the jax and numpy APIs more consistent, I added a `StochasticIntegrator` base class in 86a5f18. This allows explicitly passing an RNG instance to `multiple_steps` for determinism when using the non-jax API.